internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/issuecreator/JIRAIssueCreator.java

```diff
@@ -0,0 +1,242 @@
+                    log.info("[JS_SEC_DAILY_SCAN] JIRA Comment added : " + createComment(entry.getKey()));
+                    jira.invokePutMethodWithFile(auth, IssueCreatorConstants.WSO2_JIRA_BASE_URL +
+                            IssueCreatorConstants.JIRA_REST_API_URL + "/" +
+                            issueKey + "/attachments", fileMapper.get(entry.getKey()));
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Create JIRA Issue.
+     *
+     * @param product  product name
+     * @param filePath File path where the generated report is located. It will attached with ticket.
+     * @throws IssueCreatorException exception occurred while creating issue ticket.
+     */
+    private void createJIRAIssue(String product, String
+            filePath, String auth, String summary) throws IssueCreatorException {
+        String issue;
+        String assignee = null;
+        if (product.contains(JSScannerConstants.AM)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.AM);
+        } else if (product.contains(JSScannerConstants.INTEGRATION)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.INTEGRATION);
+        } else if (product.contains(JSScannerConstants.STREAMPROCESSOR)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.STREAMPROCESSOR);
+        } else if (product.contains(JSScannerConstants.IDENTITYSERVER)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.IDENTITYSERVER);
+        } else if (product.contains(JSScannerConstants.OB)) {
+            assignee = this.getAssigneeMapper().get(JSScannerConstants.OB);
+        }
+
+        if (assignee == null) {
+            assignee = IssueCreatorConstants.DEFAULT_ASSIGNEE;
+        }
+
+        String jiraBaseUrl = IssueCreatorConstants.WSO2_JIRA_BASE_URL;
+
+        String issueTobeCreated = createNewTicket(product, assignee, summary);
+        try {
+            issue = jira.invokePostMethod(auth, jiraBaseUrl + IssueCreatorConstants.JIRA_REST_API_URL,
+                    issueTobeCreated);
+            log.info("[JS_SEC_DAILY_SCAN] Issue ticket for " + product + " successfully created. Please find issue " +
+                    "detail : ");
+            JSONObject createdIssue = new JSONObject(issue);
+            String issueKey = createdIssue.getString("key");
+            jira.invokePostComment(auth, jiraBaseUrl + IssueCreatorConstants.JIRA_REST_API_URL +
+                            "/" +
+                            issueKey + "/comment",
+                    createComment(product));
+            log.info("[JS_SEC_DAILY_SCAN] JIRA Comment added ");
+            jira.invokePutMethodWithFile(auth, jiraBaseUrl + IssueCreatorConstants.JIRA_REST_API_URL + "/" +
+                    issueKey + "/attachments", filePath);
+        } catch (AuthenticationException e) {
+            throw new IssueCreatorException("Failed to authenticate JIRA : " + e.getMessage());
```

Comment:
Throwing just the message should not be enough. The root exception should be also thrown in order to help troubleshooting errors. This will require adding additional constructors to your exception classes.

Reply:
Resolved